packages/derived-wallet-ethereum/src/EIP1193DerivedWallet.ts

```diff
@@ -219,6 +219,7 @@ export class EIP1193DerivedWallet implements AptosWallet {
     return signAptosMessageWithEthereum({
       eip1193Provider: this.eip1193Provider,
       authenticationFunction: this.authenticationFunction,
+      aptosChainId: NetworkToChainId[this.defaultNetwork],
       messageInput: {
         ...input,
         chainId,
```

Comment:
Let's do something like this maybe?

```typescript
const chainId = NetworkToChainId[this.defaultNetwork];
return signAptosMessageWithEthereum({
  eip1193Provider: this.eip1193Provider,
  authenticationFunction: this.authenticationFunction,
  chainId,
  messageInput: {
    ...input,
    chainId: input.chainId ? chainId : undefined,
  },
});
```

Reply:
Alternatively we could just force the message to include the chain id, that might be simpler.

```typescript
const chainId = NetworkToChainId[this.defaultNetwork];
return signAptosMessageWithEthereum({
  eip1193Provider: this.eip1193Provider,
  authenticationFunction: this.authenticationFunction,
  messageInput: {
    ...input,
    chainId,
  },
});
```

and then change the chainId to required in the params for `signAptosMessageWithEthereum`

Reply:
whichever you prefer